packages/babel-generator/src/generators/types.ts

```diff
@@ -96,14 +118,16 @@ export function ArrayExpression(this: Printer, node: t.ArrayExpression) {
     if (elem) {
       if (i > 0) this.space();
       this.print(elem);
-      if (i < len - 1) this.token(",");
+      if (i < len - 1 || this.shouldPrintTrailingComma("]")) {
+        this.token(",", false, i);
```

Comment:
Q1: For each comma, we initiate a linear token scan from the start of the array expression, doesn't this imply an O(#element^2) search over the tokens array? Can we somehow reuse the token index of the previously printed comma?

Q2: Should this change be also cherry-picked to the `TupleExpression` generator?